src/components/TeacherComponents/LessonOverviewCreateWithAiDropdown/teachingMaterialsConfig.ts

```diff
@@ -0,0 +1,249 @@
+    ],
+  },
+
+  // Religious Education - no comprehension for KS1
+  "religious-education": {
+    ks1: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+
+  // Computing - all materials allowed
+  computing: {
+    ks1: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+
+  // RSHE/PSHE - all materials allowed
+  "rshe-pshe": {
+    ks1: [
+      "additional-glossary",
+      "additional-comprehension",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+
+  // Financial Education - no comprehension for any key stage
+  "financial-education": {
+    ks1: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+    ks2: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+    ks3: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+    ks4: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+
+  // Modern Foreign Languages (Spanish/French/German) - starter/exit quizzes only
+  spanish: {
+    ks2: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks3: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks4: ["additional-starter-quiz", "additional-exit-quiz"],
+  },
+
+  french: {
+    ks2: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks3: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks4: ["additional-starter-quiz", "additional-exit-quiz"],
+  },
+
+  german: {
+    ks2: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks3: ["additional-starter-quiz", "additional-exit-quiz"],
+    ks4: ["additional-starter-quiz", "additional-exit-quiz"],
+  },
+
+  // Cooking and Nutrition - all materials allowed
+  "cooking-nutrition": {
+    ks1: [
+      "additional-glossary",
+      "additional-comprehension",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+
+  // Music - all materials allowed
+  music: {
+    ks1: [
+      "additional-glossary",
+      "additional-comprehension",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  },
+};
+
+export function getTeachingMaterialTypesByCategory(
+  categories: Array<string | number | null>,
+): TeachingMaterialType[] | undefined {
+  const subjectCategoryAvailableTeachingMaterialTypes: Record<
+    string,
+    TeachingMaterialType[]
+  > = {
+    Handwriting: ["additional-glossary", "additional-starter-quiz"],
+    Spelling: [
+      "additional-glossary",
+      "additional-starter-quiz",
+      "additional-exit-quiz",
+    ],
+  };
+
+  for (const category of categories) {
+    if (subjectCategoryAvailableTeachingMaterialTypes[String(category)]) {
+      return subjectCategoryAvailableTeachingMaterialTypes[String(category)];
+    }
+  }
```

Comment:
I wonder if this could be tidier with an Array.find. But this is completely fine